### DISCUSSION TITLE
Handling cache data sorting for infinite query results

### DISCUSSION THREAD

**Discussion by swikars1** [TanStack/query#6725] [Q&A]
I was wondering what was the best was the best way to sort the cached data of infinite query results. For eg: 
```
// this is initial data of infinite query in cache
{pages: [ [{id:1; order 3}, {id:2; order 1}],  [{id:3; order 2}, {id:4; order 5}] ]}

// this is the sorted data (sorted by 'order' key)
{pages: [ [{id:2; order 1}, {id:3; order 2}],  [{id:1; order 3}, {id:4; order 5}] ]}
```

**Comment by TkDodo** [comment]
definitely create a copy of it and then sort it in the UI, or in `select` if you want. `remeda` has a nice `sortBy` util and is immutable per default:

https://remedajs.com/docs#sortBy

so:

```
sortBy(data.pages, page => page.order)
``

should work
